File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        Stack<int[]> st = new Stack<>();

        for (int i=0; i<temperatures.length; i++) {
            int temperature = temperatures[i];
            while (!st.isEmpty() && st.peek()[0] < temperature) {
                int[] temp = st.pop();
                res[temp[1]] = i - temp[1];
            }
            st.push(new int[]{temperature,i});
        }
        return res;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for t in tokens:
            if t == '+':
                num2 = int(stack.pop())
                num1 = int(stack.pop())
                t = num1 + num2
            elif t == '-':
                num2 = int(stack.pop())
                num1 = int(stack.pop())
                t = num1 - num2
            elif t == '*':
                num2 = int(stack.pop())
                num1 = int(stack.pop())
                t = num1 * num2
            elif t == '/':
                num2 = int(stack.pop())
                num1 = int(stack.pop())
                t = num1 / num2
            stack.append(int(t))

        return stack.pop()

        